apis/core/v1alpha2/readiness_types.go

```diff
@@ -7,22 +7,48 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // ReadinessSpec defines the desired state of Readiness
 type ReadinessSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Checks is the set of checks that are required to mark the readiness
+	Checks []Check `json:"checks"`
+}
+
+type Check struct {
+	// Name is the name of the check
+	Name string `json:"name"`
 
-	// Foo is an example field of Readiness. Edit readiness_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Type is the type of the check. the Type can be either basic or composite
+	// +kubebuilder:validation:Enum=basic;composite
+	Type string `json:"type"`
+
+	// Category is the category of the check. Examples of catagories are availability and security.
+	Category string `json:"category"`
 }
 
 // ReadinessStatus defines the observed state of Readiness
 type ReadinessStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	CheckStatus      []CheckStatus `json:"checkStatus"`
+	Ready            bool          `json:"ready"`
```

Comment:
A comment explaining how `Ready` can be true or false. It'd also be good to mention that this is not the **live** state , and is actually the state at `LastComputedTime`

Reply:
Done